SCons/Taskmaster.py

```diff
@@ -471,13 +471,15 @@ def postprocess(self):
                         if p.ref_count == 0:
                             self.tm.candidates.append(p)
 
+        new_candidates = []
         for p, subtract in parents.items():
             p.ref_count = p.ref_count - subtract
             if T: T.write(self.trace_message('Task.postprocess()',
                                              p,
                                              'adjusted parent ref count'))
             if p.ref_count == 0:
-                self.tm.candidates.append(p)
+                new_candidates.append(p)
+        self.tm.candidates.extend(sorted(new_candidates, key = lambda c: len(c.waiting_parents)))
```

Comment:
Wouldn't it be better to append, then sort the whole array by # of parents?

Reply:
The `candidates` list can be very long. I don't think I'd want to pay to re-sort it each time we completed a node. The goal of this change is just to ensure that if post-processing a new node unblocks several others that we stage the newly unblocked node that will itself unblock the most work at the top of the `candidates` stack.

We could investigate whether using a more sophisticated data structure could be used here, but I'd want to do that as a separate PR. At minimum, it would be important to consider what effects that might have on `Random`. The current approach implements `Random` by randomizing the order in which things are added to the `candidates` list, but if that list is something like a priority queue where "number of things unblocked" is the prioritization, that'd mean that the insertion order doesn't matter. There is also the issue that the "number of things unblocked" is adjusted dynamically, since we may not yet have evaluated all the candidates that feed into that metric.